google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -267,12 +266,8 @@ private File doStageCredentials(String googleUsername) {
     );
     String jsonCredential = new Gson().toJson(credentialMap);
     try {
-      credentialsPath = FileUtil
-          .createTempFile(
-              "tmp_google_application_default_credential",
-              "json",
-              true /* deleteOnExit */);
-      Files.write(jsonCredential, credentialsPath, Charset.forName("UTF-8"));
+      credentialsPath = Files.createTempFile("tmp_google_application_default_credential", "json");
```

Comment:
I think the default for temp files is to delete on exit. (i.e., a behaviour similar to File.deleteOnExit())
I just tested a deployment and the credentials file is deleted after the deployment thread is done.